test/rules/std_AL09_CP02_RF06_combo_test.py

```diff
@@ -0,0 +1,194 @@
+"""Test the interactions of AL09, CP02 & RF06.
+
+AL09: Self aliasing
+CP02: Identifier Capitalisation
+RF06: Identifier Quoting
+"""
+
+import pytest
+
+import sqlfluff
+
+input_query = """
+select
+    a as A,
+    B as b,
+    "C" as C,
+    "d" as d,
+    "E" as e,
+    f as F,
+    g as "G",
+    h as h,
+    I as I
+from foo
+"""
+
+
+@pytest.mark.parametrize(
+    "rules,dialect,fixed_sql",
+    [
+        # NOTE: The first few examples here are with ANSI which is
+        # configured as a natively UPPERCASE dialect.
+        (
+            ["AL09"],
+            "ansi",
+            """
+select
+    a as A,
+    B as b,
+    "C" as C,
+    "d" as d,
+    "E" as e,
+    f as F,
+    g as "G",
```

Comment:
If `CP02` is not enabled, what is the reason that `a as A` and `B as b` are not considered self-alias?

Reply:
Good question.

Here's my thinking:
- Someone has disabled `CP02`. They've decided that they don't want SQLFluff to complain if they use inconsistent casing for unquoted identifiers. That implies the _want_ to be specific in the casing of unquoted identifiers. Remember that this question is only relevant if someone has made an active choice to not enable `CP02`.
- If someone has decided they want to be specific in their identifier quoting, then if we consider `a as A` and `B as b` to be unnecessary, we're effectively disagreeing with that choice.

In some dialects this would be inappropriate all the time (clickhouse, which is always case sensitive), or in others it's _configurably inappropriate_ (TSQL, can be configured to always be case sensitive).

Do you think we should be more aggressive for at least some dialects even if CP02 is off?

Reply:
I think we've got the data on enough dialects now that we _could_ do it if we wanted (exclude the ones we know are problematic, and otherwise correct to the alias). What I'm thinking more is whether we _should_.

Reply:
Thanks for the explanation! I agree that it could be an intentional behaviour to disable `CP02` to have the specific casing. 

However, my thinking is that we would not know if this is intentional behaviour by the user to do `a as A` and `B as b`, or an error. Since `AL09` is enabled, I believe we should still mark these errors as _unfixable_ lint errors. Then the user has the option to quote the alias, or disable the rule `AL09` for these lines or entirely. This way a user can specify their intended behaviour.

What do you think?

Reply:
I like that suggestion 👍 . I think I was getting hung up on the right way to fix it, but you're right that it should still be an error, just one that the user will need to resolve.